File: src/1009_complement_of_base_10_integer/ComplementOfBase10Integer.java
class Solution {
    public int bitwiseComplement(int n) {
        if (n == 0) {
            return 0;
        }

        int value = n;
        int mask = 0;

        while (value > 0) {
            mask = (mask << 1) | 1;
            value >>= 1;
        }

        return mask ^ n;
    }
}

public class ComplementOfBase10Integer {
    private static void runTests() {
        Solution sol = new Solution();
        int[] inputs = { 5, 7, 10, 0, 1, 8 };
        int[] expecteds = { 2, 0, 5, 1, 0, 7 };

        for (int i = 0; i < inputs.length; i++) {
            int actual = sol.bitwiseComplement(inputs[i]);
            System.out.println(actual + " | " + expecteds[i]);
        }
    }

    public static void main(String[] args) {
        runTests();
    }
}

File: src/1009_complement_of_base_10_integer/complement_of_base_10_integer.py
from typing import List


class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1

        mask = 0
        value = n
        while value > 0:
            mask = (mask << 1) | 1
            value >>= 1

        return n ^ mask


def run_tests() -> None:
    sol = Solution()
    inputs: List[int] = [5, 7, 10, 0, 1, 8]
    expecteds: List[int] = [2, 0, 5, 1, 0, 7]

    for i, n in enumerate(inputs):
        actual = sol.bitwiseComplement(n)
        print(f"{actual} | {expecteds[i]}")


if __name__ == "__main__":
    run_tests()
